src/accounts.rs

```diff
@@ -337,9 +337,14 @@ impl Config {
 
     /// Sync the inmemory representation to disk.
     async fn sync(&self) -> Result<()> {
-        fs::write(&self.file, toml::to_string_pretty(&self.inner)?)
+        let file_tmp = self.file.with_extension("toml.tmp");
```

Comment:
In the current code calling `sync()` in parallel is only possible when creating multiple `Accounts` objects which is wrong by design. I think it's better to try keeping this invariant on the type system level. Otherwise if we have multiple `Accounts`/`Config` obejcts, there are more complex problems coming like synchronisation of them